htdocs/core/class/commonobject.class.php

```diff
@@ -4744,6 +4744,11 @@ function fetch_optionals($rowid=null, $optionsArray=null)
 								//var_dump($extrafields->attributes[$this->table_element]['type'][$key]);
 								$this->array_options["options_".$key]=$this->db->jdate($value);
 							}
+							// If field is a computed field, value must become result of compute
+							else if (! empty($extrafields) && !empty($extrafields->attributes[$this->table_element]['computed'][$key]))
```

Comment:
Image the computed field of the sum of 2 other following extra fields (field a = b+c).
Here you make the dol_eval before the $this->array_options["options_b"] and $this->array_options["options_c"] is defined.
Solution may be to add a second loop (first ignore computed field, second one set only computed fields)